File: src/phase2/recursion/submissions/JosephusProblem.java
package phase2.recursion.submissions;

public class JosephusProblem {
    public static void main(String[] args) {
      System.out.println(josephus(3,2));
    }

    public static int josephus(int n, int k)
    {
        if (n == 1) {
            return 0;
        }
        return (josephus(n-1, k)+k)%n;
    }

}

File: src/phase2/recursion/JosephusProblem.kt
package phase2.recursion

fun jProblem(n: Int, k: Int): Int {
    if (n == 1) {
        return 0
    } else {
        return (jProblem(n-1, k)+k)%n
    }
}

fun main() {
    print(jProblem(5, 2))
}